src/kirby/components/floating-action-button/floating-action-button.component.tns.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, NgZone, Input } from '@angular/core';
+import { OrientationChangedEventData } from 'tns-core-modules/application';
+import { ContentView } from 'tns-core-modules/ui/content-view';
+import { EventData, View } from 'tns-core-modules/ui/page/page';
+import * as app from 'tns-core-modules/application';
+import { Color } from 'tns-core-modules/color';
+import { registerElement } from 'nativescript-angular';
+
+const FLOATING_ACTION_BUTTON_COMPONENT_SELECTOR = 'kirby-floating-action-button';
+
+declare const CGSizeMake: any;
+declare const android: any;
+declare var require: any;
+const style: any = require('sass-extract-loader!./floating-action-button.component.scss');
+
+@Component({
+  selector: FLOATING_ACTION_BUTTON_COMPONENT_SELECTOR,
+  templateUrl: './floating-action-button.component.html',
+  styleUrls: ['./floating-action-button.component.scss'],
+})
+export class FloatingActionButtonComponent extends ContentView {
+  @Input() iconSrc?: string = '~/assets/icons/add/add@3x.png';
+  @Input() hasShadow?: boolean = true;
+  @Input() disabled?: boolean = false;
+
+  view: View;
+
+  constructor(private zone: NgZone) {
+    super();
+  }
+
+  onViewLoaded(args: EventData) {
+    this.view = <View>args.object;
+    this.setupOnOrientationChangeListener();
+    this.addShadow();
+  }
+
+  setupOnOrientationChangeListener() {
+    app.on(app.orientationChangedEvent, (args: OrientationChangedEventData) => {
+      // Run in the zone, to make sure Angular data binding is informed of this:
+      this.zone.run(() => this.addShadow());
+    });
+  }
+
+  addShadow(): void {
```

Comment:
This looks somewhat similar to the code of the `addShadow()`-method in `card.component.tns.ts`, maybe this can be generalized upon... or maybe the similarities are simply superficial... have you considered if the code can be generalized upon?

Reply:
Completely agree with you. We also discussed the same with Jakob last week and agreed to keep it static for the time being.

Reply:
Noted...accepted... albeit not intuitive when reviewing, consider adding a "TODO"-comment (although those have a tendency of never being revisited, I doubt that'll be the case here).